featuretools/computational_backends/feature_set_calculator.py

```diff
@@ -936,7 +939,16 @@ def update_feature_columns(feature_data, data):
         data.update(new_cols)
         return data
 
-    return data.assign(**new_cols)
+    if isinstance(data, pd.DataFrame):
+        return pd.concat([data, pd.DataFrame(new_cols, index=data.index)], axis=1)
+    else:
+        for name, col in new_cols.items():
+            col.name = name
+            if isinstance(data, dd.DataFrame):
+                data = dd.concat([data, col], axis=1)
+            else:
+                data = ps.concat([data, col], axis=1)
+        return data
```

Comment:
I don't really like this approach for Dask/Spark, but we can't create a new dataframe from the columns here like we can for pandas, so we have to concat the columns one at a time. Maybe there is a better way, but I didn't come up with one off the top of my head.